### ISSUE TITLE
Deserializing `BigDecimal` with large number of decimals result in incorrect value

### ISSUE THREAD

**Issue by lnthai2002** [FasterXML/jackson-databind#4694]
### Search before asking

- [X] I searched in the [issues](https://github.com/FasterXML/jackson-databind/issues) and found nothing similar.

### Describe the bug

I have a string represents a number with 652 decimal places (-11000.00000...). When reading this string to BigDecimal i end up with a wrong number -1.1000E-648. The simple example is found at https://github.com/lnthai2002/jackson-databind-test/blob/db3e060ac38c555b001d91c549e004a19f9a17c7/src/main/java/App.java

### Version Information

2.17.1

### Reproduction

<-- Any of the following
1. Brief code sample/snippet: include here in preformatted/code section
2. Longer example stored somewhere else (diff repo, snippet), add a link
3. Textual explanation: include here
 -->
```java
import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.JsonNodeFactory;

import java.math.BigDecimal;

public class App {
  public static void main (String [] args) {
    JsonNodeFactory f = JsonNodeFactory.withExactBigDecimals(true);
    ObjectMapper mapper = new ObjectMapper();
    mapper.findAndRegisterModules()
        .setNodeFactory(f)
        .configure(JsonGenerator.Feature.WRITE_BIGDECIMAL_AS_PLAIN, true)
        .configure(DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS, true);
    String str = "-11000.0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000";
    try {
      BigDecimal num = mapper.readValue(str, BigDecimal.class);
      System.out.println(num);
    } catch (Exception e) {
      throw new RuntimeException(e);
    }
  }
}
``` 

### Expected behavior

should print "-11000.0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"

### Additional context

If i switch back to 2.14.1 then i got the correct result

**Comment by pjfanning** [comment]
@cowtowncoder the issue is in FastDoubleParser `JavaBigDecimalParser.parseBigDecimal(arr, 0, arr.length)`.

`JavaBigDecimalParser.parseBigDecimal(str)` handles the number fine but the char array method has a bug.

In Jackson 2.17, the BigDecimalParser uses the FastDoubleParser for BigDecimal when there are more than 500 chars.

I can report the issue to FastDoubleParser and also look at changing BigDecimalParser to work around the issue.

**Comment by cowtowncoder** [comment]
Ok. Let's first see if we could get a fix for FDP... Ideally would not want to add temporary work-arounds, esp. given we are shading the dependency so can relatively freely upgrade.

I think I can add a failing test here, for 2.17, and we can see what to do (workaround vs upgrade) in couple of days.

**Comment by cowtowncoder** [comment]
Fixed in 2.17 (for 2.17.3) and 2.18 (for 2.18.0) -- fix actually via `jackson-core`, databind does have passing tests tho.

**Comment by lnthai2002** [comment]
there is no jackson-bom 2.17.2 ?

**Comment by pjfanning** [comment]
> there is no jackson-bom 2.17.3 ?

@lnthai2002 why are commenting on random closed issues? Jackson 2.17.3 has not been released.

**Comment by cowtowncoder** [comment]
As per @pjfanning 2.17.3 is not yet released and it will take a while for it to be since there are just a couple of fixes, as per:

https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.17.3

and full patch release takes 2-3 hours to do. So this is not done for single bug fixes.
